lib/carrierwave/storage/aws_file.rb

```diff
@@ -93,10 +93,11 @@ def public_url
       def url(options = {})
         if signer
           signed_url(options)
-        elsif uploader.aws_acl.to_s != 'public-read'
-          authenticated_url(options)
-        else
+        elsif uploader.aws_acl.to_s == 'public-read' ||
```

Comment:
This clause is rather lengthy. It would be nice to have it extracted into a private helper method, just for clarity.